carp.common/src/commonMain/kotlin/dk/cachet/carp/common/users/ParticipantAttribute.kt

```diff
@@ -0,0 +1,100 @@
+package dk.cachet.carp.common.users
+
+import dk.cachet.carp.common.NamespacedId
+import dk.cachet.carp.common.UUID
+import dk.cachet.carp.common.data.Data
+import dk.cachet.carp.common.data.input.AnyInputElement
+import dk.cachet.carp.common.data.input.CUSTOM_INPUT_TYPE
+import dk.cachet.carp.common.data.input.CustomInput
+import dk.cachet.carp.common.data.input.InputElement
+import dk.cachet.carp.common.data.input.InputDataType
+import dk.cachet.carp.common.data.input.InputDataTypeList
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+
+
+/**
+ * Describes expected data related to one or multiple participants in a study.
+ */
+@Serializable
+sealed class ParticipantAttribute
+{
+    /**
+     * Uniquely identifies the type of data represented by this participant attribute.
+     */
+    abstract val inputType: InputDataType
+
+    /**
+     * The unique identifier for this participant attribute.
+     */
+    abstract val id: NamespacedId
```

Comment:
Is it just me or is only one ever needed? The name and namespace is simply only set to a unique value in the case of the custom attribute.

Reply:
Both are needed. All custom input is of type `CUSTOM_INPUT_TYPE`, which is required to be able to evaluate whether or not this is custom input in `inputToData`

The `id` serves a different purpose and just so happens to overlap for the default types (why use yet a different `NamespaceId`?)

Reply:
or all custom input is in the _namespace_ `CUSTOM_INPUT_TYPE` and has a unique name (from uuid) which btw is how the id is implemented. The use of id is a bit confusing since it is just the data type after all. It needs to be bette documented exactly why you need both at least

Reply:
I considered that, but then we can't use `CUSTOM_INPUT_TYPE_NAME` as custom `@SerialName`.

But, maybe we don't have to ...